modules/ROOT/pages/_partials/commons/common-new-logging-api.adoc

```diff
@@ -0,0 +1,634 @@
+
+----
+--
+
+Kotlin::
++
+--
+
+.Old API
+:!example-caption:
+[source, kotlin, indent=0]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/BasicExamples.kt[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, kotlin, indent=0]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/BasicExamples.kt[tags=new-file-logging]
+
+----
+--
+=====
+
+endif::is-android[]
+ifdef::is-java[]
+
+.Old API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=new-file-logging]
+
+----
+
+endif::is-java[]
+ifdef::is-csharp[]
+
+.Old API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=file-logging]
+----
+
+.New API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=new-file-logging]
+----
+endif::is-csharp[]
+ifdef::is-objc[]
+
+.Old API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=new-file-logging]
+
+----
+endif::is-objc[]
+ifdef::is-swift[]
+
+.Old API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=new-file-logging]
+
+----
+endif::is-swift[]
+ifdef::is-c[]
+
+.Old API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=new-file-logging]
+
+----
+endif::is-c[]
+
+[#lbl-custom-logsink]
+=== Using a Custom Logger
+
+Installing a custom log sink with the new API is also streamlined: create an
+instance of your custom sink and use `LogSinks.get().setCustom` to install it.
+
+As with the other log sinks, you will have to specify the level and domain
+at which Couchbase logs are forwarded to your custom sink at its creation.
+
+Your custom log sink code will have to change as well.  The most significant
+change is that instead of _implementing_ the `Logger` interface, a custom log sink
+must _extend_ the `BaseLogSink` class.  
+
+The `BaseLogSink` class does not have a no-args constructor: you must specify at least the level at which Couchbase logs will be forwarded to the logger.  
```

Comment:
Not sure what is the best way to describe this in the cross-platform way. The `BaseLogSink` described here only applies to Java/Android and .NET. 

In Swift and Objective-C, which lack abstract methods, a custom log sink implements LogSinkProtocol and is assigned to a CustomLogSink instance during creation.

Similarly, in C, the custom log sink implementation is a callback that is assigned to CBLCustomLogSink object during creation.

Reply:
I think that for the custom log sink every platform should have its own rubric with a brief explanation and the specifics. There are snippets from old to new api just below that are platform specific as well and so it should suffice.

Reply:
Done